tools/migration.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+from teamscale_client import TeamscaleClient
+from abc import ABC, abstractmethod
+
+OLD_TEAMSCALE = {
+    "url": "http://localhost:8080",
+    "project": "old",
+    "user": "user",
+    "token": "tokentoken"
+}
+
+NEW_TEAMSCALE = {
+    "url": "http://localhost:8080",
+    "project": "new",
+    "user": "user",
+    "token": "tokentoken"
+}
+
+
+def main():
+    client_new = TeamscaleClient(NEW_TEAMSCALE["url"],
+                                 NEW_TEAMSCALE["user"],
+                                 NEW_TEAMSCALE["token"],
+                                 NEW_TEAMSCALE["project"])
+
+    client_old = TeamscaleClient(OLD_TEAMSCALE["url"],
+                                 OLD_TEAMSCALE["user"],
+                                 OLD_TEAMSCALE["token"],
+                                 OLD_TEAMSCALE["project"])
+
+    BlacklistMigrator(client_old, client_new).migrate()
+    TaskMigrator(client_old, client_new).migrate()
+
+
+class MigratorBase(ABC):
+    """ Base class for migrating data from one instance to another via REST calls. """
+
+    def __init__(self, old, new):
+        self.old = old
+        self.new = new
+        self.migrated = 0
+
+    @staticmethod
+    def get(client, service, path_param="", parameters=None):
+        """ Performs a GET call from the client to the service with the given parameters
+        and returns the response as a JSON Object.
+        The path parameter are additions to the path, e.g /service/id.
+        """
+        response = client.get(client.get_project_service_url(service) + path_param, parameters)
+        return json.loads(response.text)
```

Comment:
It can be done. At the moment though, I abort everytime a ServiceError is raised. I think we need to check at the MR what problems might arise and if there is actually a use case in which resuming the migration with a 404 make sense. It would add a lot of None checks throughout the migrators and I rather not to this preemptively